mysite/qq_config.txt
[root@localhost VideoSpider]# python starter.py qq SeedsSpider
★STEP INFO★: LOG_FILE = /home/TaveenSpider_log/logs/qq/[SeedsSpider] 2016-06-16 17.04.21.837887/qq_seeds.log
★STEP INFO★: 启动 starter.py @ 2016-06-16 17:04:21.837887
★STEP INFO★: 解析命令行启动参数...
★STEP INFO★: Website = qq
★STEP INFO★: SpiderType = SeedsSpider
★STEP INFO★: 加载种子爬虫模板(qq_seeds.tmplt)...
★TEMPLATE ★: > tv = {"video_url_regex": [], "filter_url_pattern": "http://v.qq.com/list/2_[#类型#]_[#地区#]_[#年份#]_0_0_[#PAGE#]_20_-1_-1_0.html>", "page_count": {"pattern": "//div[@id='pager']//span[@class='mod_pagenav_count']//a[last()]//text()", "extract_type": "xpath", "start_page_count": 0}, "base_url": "http://v.qq.com/list/2_-1_-1_-1_0_0_[#PAGE#]_20_-1_-1_0.html", "filter": {"地区": {"dont_filter_value": "-1", "extract_type": "enum", "data": "-1|0|8|1|4|5|9999"}, "类型": {"dont_filter_value": "-1", "extract_type": "enum", "data": "-1|1|2|3|4|5|6|7|8|9|10|11|12|13|14|15|16|17|18|19|20|21|22|23|24"}, "年份": {"dont_filter_value": "-1", "extract_type": "enum", "data": "-1|2014|2013|2012|2011|2010|2009|2008|2007|2006|9999"}}, "video_info": {"extract_type": "xpath", "video_url": {"path": ".//h6[@class='scores']/a/@href", "concat": ""}, "title": {"path": ".//h6[@class='scores']/a/text()", "concat": ""}, "pay": {"path": ".//sup[@class='mark_vip_coupon' or @class='mark_vip_free']/text()", "concat": ""}, "image": {"path": ".//img/@src", "concat": ""}, "base_tag": "//div[@id='content']//ul//li", "actor": {"path": ".//p[contains(text(), '主演')]//a/text()", "concat": ""}, "director": {"path": ".//p[contains(text(), '导演')]//a/text()", "concat": ""}, "length": {"path": ".//em[@class='mask_txt']/text()", "concat": ""}, "curEpisode": {"path": ".//em[@class='mask_txt']/text()", "concat": ""}}}
★TEMPLATE ★: > movie = {"video_url_regex": [], "filter_url_pattern": "http://v.qq.com/movielist/1/[#类型#]_[#地区#]_[#年份#]_0_0/0/0/[#PAGE#]/20/0/0.html", "page_count": {"pattern": "//div[@id='pager']//span[@class='mod_pagenav_count']//a[last()]//text()", "extract_type": "xpath", "start_page_count": 0}, "base_url": "http://v.qq.com/movielist/10001/0/0/0/[#PAGE#]/20/0/0.html", "filter": {"地区": {"dont_filter_value": "0", "extract_type": "enum", "data": "0|10002-100024|10002-100025|10002-100026|10002-100027|10002-100028|10002-100029|10002-100030|10002-100031|10002-100032|10002-100033"}, "类型": {"dont_filter_value": "0", "extract_type": "enum", "data": "0|10001-100061|10001-100003|10001-100004|10001-100005|10001-100006|10001-100007|10001-100008|10001-100009|10001-100010|10001-100011|10001-100012|10001-100013|10001-100014|10001-100015|10001-100016|10001-100017|10001-100018|10001-100019|10001-100020|10001-100021|10001-100022|10001-100062"}, "年份": {"dont_filter_value": "-1", "extract_type": "enum", "data": "0|10003-100034|10003-100035|10003-100036|10003-100037|10003-100038|10003-100039|10003-100040"}}, "video_info": {"extract_type": "xpath", "video_url": {"path": ".//h6[@class='scores']/a/@href", "concat": ""}, "title": {"path": ".//h6[@class='scores']/a/text()", "concat": ""}, "pay": {"path": ".//sup[@class='mark_vip_coupon' or @class='mark_vip_free']/text()", "concat": ""}, "image": {"path": ".//img/@src", "concat": ""}, "base_tag": "//div[@id='content']//ul//li", "actor": {"path": ".//p[contains(text(), '主演')]//a/text()", "concat": ""}, "director": {"path": ".//p[contains(text(), '导演')]//a/text()", "concat": ""}, "length": {"path": ".//em[@class='mask_txt']/text()", "concat": ""}, "curEpisode": {"path": ".//em[@class='mask_txt']/text()", "concat": ""}}}
★TEMPLATE ★: > variety = {"video_url_regex": [], "filter_url_pattern": "http://v.qq.com/variety/type/list_[#归属电视台OR节目类型#]_0_[#PAGE#].html", "page_count": {"pattern": "//div[@id='pager']//span[@class='mod_pagenav_count']//a[last()]//text()", "extract_type": "xpath", "start_page_count": 0}, "base_url": "http://v.qq.com/variety/type/list_-1_0_[#PAGE#].html", "filter": {"归属电视台OR节目类型": {"dont_filter_value": "-1", "extract_type": "enum", "data": "-1|51|10|2|3|4|5|77|6|75|7|9|11|12|34|35|13|72|39|14|38|70|71|44|50|49|47|40|41|42|66|37|45|46|76|74|64|61|60|56|54|63|73|36|15|17|18|19|20|21|22|59|23|32|24|33|68|67|69|25|26|65|29|28|57|58"}}, "video_info": {"extract_type": "xpath", "video_url": {"path": ".//div[@class='mod_item_tit']//a/@href", "concat": ""}, "title": {"path": ".//div[@class='mod_item_tit']//a/text()", "concat": ""}, "pay": {"path": ".//sup[@class='mark_vip_coupon' or @class='mark_vip_free']/text()", "concat": ""}, "image": {"path": ".//div[@class='mod_pic']//a/img/@src", "concat": ""}, "base_tag": "//div[@class='mod_video_list details']//div[@class='mod_cont']//div[@class='mod_item pic_160']", "length": {"path": ".//div[@class='mod_pic']//span[@class='figure_mask']//em[@class='mask_txt']/text()", "concat": ""}, "quality": {"path": ".//div[@class='mod_pic']//sup[@class='mark_sd' or @class='mark_hd']/text()", "concat": ""}, "curEpisode": {"path": ".//div[@class='mod_pic']//span[@class='figure_mask']//em[@class='mask_txt']/text()", "concat": ""}}}
★TEMPLATE ★: > animation = {"video_url_regex": [], "filter_url_pattern": "http://v.qq.com/cartlist/[#PAGE_MOD_10#]/3_[#类型#]_[#地区#]_[#年龄#]_[#版本#]_0_[#PAGE#]_0_20.html", "page_count": {"pattern": "//div[@id='pager']//span[@class='mod_pagenav_count']//a[last()]//text()", "extract_type": "xpath", "start_page_count": 0}, "base_url": "http://v.qq.com/cartlist/[#PAGE_MOD_10#]/3_-1_-1_-1_-1_0_[#PAGE#]_0_20.html", "filter": {"地区": {"dont_filter_value": "-1", "extract_type": "enum", "data": "-1|0|1|2|3"}, "类型": {"dont_filter_value": "-1", "extract_type": "enum", "data": "-1|0|1|2|3|4|5|6|7|8|9|10|11"}, "年龄": {"dont_filter_value": "-1", "extract_type": "enum", "data": "-1|0|1|2|3|4|5"}, "版本": {"dont_filter_value": "-1", "extract_type": "enum", "data": "-1|0|1"}}, "video_info": {"extract_type": "xpath", "video_url": {"path": ".//h6[@class='scores']/a/@href", "concat": ""}, "title": {"path": ".//h6[@class='scores']/a/text()", "concat": ""}, "pay": {"path": ".//sup[@class='mark_vip_coupon' or @class='mark_vip_free']/text()", "concat": ""}, "image": {"path": ".//img/@src", "concat": ""}, "base_tag": "//div[@id='content']//ul//li", "actor": {"path": ".//p[contains(text(), '主演')]//a/text()", "concat": ""}, "director": {"path": ".//p[contains(text(), '导演')]//a/text()", "concat": ""}, "length": {"path": ".//span[@class='mod_version']/text()", "concat": ""}, "quality": {"path": ".//span[@class='mod_HD']/text()", "concat": ""}, "curEpisode": {"path": ".//span[@class='mod_version']/text()", "concat": ""}}}
★STEP INFO★: LOG_LEVEL = DEBUG
★STEP INFO★: DOWNLOAD_DELAY = 0.0
★STEP INFO★: 启动scrapy @ 2016-06-16 17:04:21
**************************************************Spider Opened Extension**************************************************
2016-06-16 17:04:22+0800 [scrapy] INFO: Scrapy 0.24.6 started (bot: TAVEEN_SPIDER)
2016-06-16 17:04:22+0800 [scrapy] INFO: Optional features available: ssl, http11
2016-06-16 17:04:22+0800 [scrapy] INFO: Overridden settings: {'SPIDER_MODULES': ['VideoSpider.seeds_spiders', 'VideoSpider.page_spiders'], 'CONCURRENT_REQUESTS': 32, 'USER_AGENT': 'Mozilla/5.0 (Windows NT 6.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/31.0.1650.63 Safari/537.36', 'BOT_NAME': 'TAVEEN_SPIDER'}
2016-06-16 17:04:22+0800 [SeedsSpider] DEBUG: Redirecting (302) to <GET https://www.baidu.com/> from <GET http://www.baidu.com>
2016-06-16 17:04:23+0800 [SeedsSpider] DEBUG: Crawled (200) <GET https://www.baidu.com/> (referer: None)
★STEP INFO★: generator basic seeds urls(page 1)...
2016-06-16 17:04:23+0800 [SeedsSpider] DEBUG: Redirecting (301) to <GET http://v.qq.com/x/varietylist/?sort=4&offset=0> from <GET http://v.qq.com/variety/type/list_-1_0_0.html>
2016-06-16 17:04:23+0800 [SeedsSpider] DEBUG: Redirecting (meta refresh) to <GET http://v.qq.com/x/teleplaylist/?sort=4&offset=0> from <GET http://v.qq.com/list/2_-1_-1_-1_0_0_0_20_-1_-1_0.html>
2016-06-16 17:04:23+0800 [SeedsSpider] DEBUG: Redirecting (meta refresh) to <GET http://v.qq.com/x/cartoonlist/?sort=4&offset=0> from <GET http://v.qq.com/cartlist/0/3_-1_-1_-1_-1_0_0_0_20.html>
2016-06-16 17:04:23+0800 [SeedsSpider] DEBUG: Redirecting (meta refresh) to <GET http://v.qq.com/x/movielist/?cate=10001&offset=0&sort=4> from <GET http://v.qq.com/movielist/10001/0/0/0/0/20/0/0.html>
2016-06-16 17:04:23+0800 [SeedsSpider] DEBUG: Crawled (200) <GET http://v.qq.com/x/varietylist/?sort=4&offset=0> (referer: https://www.baidu.com/)
----------------------------------------------------------------------------------------------------
★9AroFpoKN★ Processing variety page -->>> http://v.qq.com/x/varietylist/?sort=4&offset=0 
★9AroFpoKN★ [0] videos added to seeds list, [0] reduplicated, [0] total.
2016-06-16 17:04:23+0800 [SeedsSpider] DEBUG: Crawled (200) <GET http://v.qq.com/x/cartoonlist/?sort=4&offset=0> (referer: https://www.baidu.com/)
----------------------------------------------------------------------------------------------------
★HWvIsuXMM★ Processing animation page -->>> http://v.qq.com/x/cartoonlist/?sort=4&offset=0 
★HWvIsuXMM★ [0] videos added to seeds list, [0] reduplicated, [0] total.
2016-06-16 17:04:23+0800 [SeedsSpider] DEBUG: Crawled (200) <GET http://v.qq.com/x/teleplaylist/?sort=4&offset=0> (referer: https://www.baidu.com/)
----------------------------------------------------------------------------------------------------
★SuEVbbNtn★ Processing tv page -->>> http://v.qq.com/x/teleplaylist/?sort=4&offset=0 
★SuEVbbNtn★ [0] videos added to seeds list, [0] reduplicated, [0] total.
2016-06-16 17:04:23+0800 [SeedsSpider] DEBUG: Crawled (200) <GET http://v.qq.com/x/movielist/?cate=10001&offset=0&sort=4> (referer: https://www.baidu.com/)
----------------------------------------------------------------------------------------------------
★YQjG6APH5★ Processing movie page -->>> http://v.qq.com/x/movielist/?cate=10001&offset=0&sort=4 
★YQjG6APH5★ [0] videos added to seeds list, [0] reduplicated, [0] total.
2016-06-16 17:04:23+0800 [SeedsSpider] INFO: Closing spider (finished)
**************************************************Spider Closed Extension**************************************************
★STATISTICS★: 统计种子数据...
★STATISTICS★: tv seeds count = 0
★STATISTICS★: movie seeds count = 0
★STATISTICS★: animation seeds count = 0
★STATISTICS★: variety seeds count = 0
★STATISTICS★: total seeds count = 0
★STEP INFO★: 写种子文件... -->>> "/home/TaveenSpider_log/output/seeds_output/qq_seeds.txt"
★STEP INFO★: 计算种子程序本次执行的相关统计信息...
★STEP INFO★: 统计完毕.
★STEP INFO★: 将种子程序本次执行的统计信息写入数据库中...
ERROR:root:Error: Failed to Connect to database:
    .
Parameters: host = 192.168.1.80, port = 3306, db = videoSpider, user = koala, pwd  = koala, charset = utf8
Make sure that all parameters have been set correctly.
Traceback (most recent call last):
  File "/root/TaveenSpider/TaveenUtil/DbOperator.py", line 74, in __connect
    passwd=self.__db_config.db_password, charset=self.__db_config.db_charset)
  File "/usr/lib64/python2.7/site-packages/MySQLdb/__init__.py", line 81, in Connect
    return Connection(*args, **kwargs)
  File "/usr/lib64/python2.7/site-packages/MySQLdb/connections.py", line 204, in __init__
    super(Connection, self).__init__(*args, **kwargs2)
OperationalError: (1049, "Unknown database 'videoSpider'")


mysite/byrbbs/byrbbs/settings.py
# coding: utf-8
# Scrapy settings for byrbbs project

LOG_LEVEL = 'INFO'

BOT_NAME = 'byrbbs'

SPIDER_MODULES = ['byrbbs.spiders']
NEWSPIDER_MODULE = 'byrbbs.spiders'

ITEM_PIPELINES = {'byrbbs.pipelines.ByrbbsPipeline': 100}

# Crawl responsibly by identifying yourself (and your website) on the user-agent
# USER_AGENT = 'byrbbs (+http://www.yourdomain.com)'
USER_AGENTS = [
  "Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.1; SV1; AcooBrowser; .NET CLR 1.1.4322; .NET CLR 2.0.50727)",
  "Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 6.0; Acoo Browser; SLCC1; .NET CLR 2.0.50727; Media Center PC 5.0; .NET CLR 3.0.04506)",
  "Mozilla/4.0 (compatible; MSIE 7.0; AOL 9.5; AOLBuild 4337.35; Windows NT 5.1; .NET CLR 1.1.4322; .NET CLR 2.0.50727)",
  "Mozilla/5.0 (Windows; U; MSIE 9.0; Windows NT 9.0; en-US)",
  "Mozilla/5.0 (compatible; MSIE 9.0; Windows NT 6.1; Win64; x64; Trident/5.0; .NET CLR 3.5.30729; .NET CLR 3.0.30729; .NET CLR 2.0.50727; Media Center PC 6.0)",
  "Mozilla/5.0 (compatible; MSIE 8.0; Windows NT 6.0; Trident/4.0; WOW64; Trident/4.0; SLCC2; .NET CLR 2.0.50727; .NET CLR 3.5.30729; .NET CLR 3.0.30729; .NET CLR 1.0.3705; .NET CLR 1.1.4322)",
  "Mozilla/4.0 (compatible; MSIE 7.0b; Windows NT 5.2; .NET CLR 1.1.4322; .NET CLR 2.0.50727; InfoPath.2; .NET CLR 3.0.04506.30)",
  "Mozilla/5.0 (Windows; U; Windows NT 5.1; zh-CN) AppleWebKit/523.15 (KHTML, like Gecko, Safari/419.3) Arora/0.3 (Change: 287 c9dfb30)",
  "Mozilla/5.0 (X11; U; Linux; en-US) AppleWebKit/527+ (KHTML, like Gecko, Safari/419.3) Arora/0.6",
  "Mozilla/5.0 (Windows; U; Windows NT 5.1; en-US; rv:1.8.1.2pre) Gecko/20070215 K-Ninja/2.1.1",
  "Mozilla/5.0 (Windows; U; Windows NT 5.1; zh-CN; rv:1.9) Gecko/20080705 Firefox/3.0 Kapiko/3.0",
  "Mozilla/5.0 (X11; Linux i686; U;) Gecko/20070322 Kazehakase/0.4.5",
  "Mozilla/5.0 (X11; U; Linux i686; en-US; rv:1.9.0.8) Gecko Fedora/1.9.0.8-1.fc10 Kazehakase/0.5.6",
  "Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/535.11 (KHTML, like Gecko) Chrome/17.0.963.56 Safari/535.11",
  "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_7_3) AppleWebKit/535.20 (KHTML, like Gecko) Chrome/19.0.1036.7 Safari/535.20",
  "Opera/9.80 (Macintosh; Intel Mac OS X 10.6.8; U; fr) Presto/2.9.168 Version/11.52",
]
